### ISSUE TITLE
Capturing Playback Audio Devices?

### ISSUE THREAD

**Issue by playmer** [libsdl-org/SDL#7691]
Someone recently mentioned needing to capture playback devices to me. They were on Windows under the WASAPI driver, and after looking into it, using the LOOPBACK functionality seemed relatively straightforward. I wrote up first attempt of what this might look like in SDL2 here: https://github.com/playmer/SDL-1/commit/0a8fe2b1f76f64139e905d5a34262d30cfbe76b5

Would this break ABI for SDL2? If so I'd be happy to make a PR for the above (I'm sure it'd need some modifications) 

Is there appetite for allowing capturing of playback devices on Drivers that can support it? If so I don't mind porting the above to SDL3.

**Comment by slouken** [comment]
This doesn't break the ABI, but it definitely feels like something that should go in SDL3. Thoughts, @icculus?

**Comment by icculus** [comment]
The goal is to capture audio from _other_ processes that are playing to that device, right?

**Comment by playmer** [comment]
Correct, capturing audio from other programs. Their use case was running voice recognition on audio from video they were playing from another process to get captions out. I tested with the sample from whisper.cpp which is what they were playing with when they noticed this wasn't working.  

And to be clear I'm happy to port this to SDL3, I just started with 2 as it's what the sample was using.
Wasn't sure if it'd go in 2, 3, neither, or both.
